FizzBuzz/part2.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+
+def FizzBuzz(the_range, rules):
+    for number in range(1, the_range + 1):
+        tokens = []
+        if number % 11 == 0 and rules[11] == True:
+            tokens.append("Bong")
+        else:
+            if number % 15 == 0 and rules[3] == True and rules[5] == True:
+                tokens.append("Fizz")
+                tokens.append("Buzz")
+            elif number % 3 == 0 and rules[3] == True:
+                tokens.append("Fizz")
+            elif number % 5 == 0 and rules[5] == True:
+                tokens.append("Buzz")
+
+            if number % 7 == 0 and rules[7] == True:
+                tokens.append("Bang")
+
+        if number % 13 == 0 and rules[13] == True:
+            where_is_b = -1
+            for tok in tokens:
+                where_is_b = tok.find("B")
+                if where_is_b != -1:
+                    index = tokens.index(tok)
+                    tokens.insert(index, "Fezz")
+                    break

```

Comment:
A nice challenge here would be to try to avoid using a `for` loop to iterate through the tokens and instead maybe use a `filter` (https://www.geeksforgeeks.org/filter-in-python/) or a `lamda` expression. Can you try doing that, but don't waste too much time if that's not working!